cmd/sovereignnode/chainSimulator/chainSimulator_test.go

```diff
@@ -0,0 +1,326 @@
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/dtos"
+)
+
+const (
+	defaultPathToInitialConfig = "../../node/config/"
+	sovereignConfigPath        = "../config/"
+)
+
+func TestNewSovereignChainSimulator(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	epochConfig, economicsConfig, sovereignExtraConfig, err := LoadSovereignConfigs(sovereignConfigPath)
+	require.Nil(t, err)
+
+	chainSimulator, err := NewSovereignChainSimulator(ArgsSovereignChainSimulator{
+		SovereignExtraConfig: *sovereignExtraConfig,
+		ChainSimulatorArgs: chainSim.ArgsChainSimulator{
+			BypassTxSignatureCheck: false,
+			TempDir:                t.TempDir(),
+			PathToInitialConfig:    defaultPathToInitialConfig,
+			NumOfShards:            1,
+			GenesisTimestamp:       time.Now().Unix(),
+			RoundDurationInMillis:  uint64(6000),
+			RoundsPerEpoch:         core.OptionalUint64{},
+			ApiInterface:           api.NewNoApiInterface(),
+			MinNodesPerShard:       2,
+			ConsensusGroupSize:     2,
+			AlterConfigsFunction: func(cfg *config.Configs) {
+				cfg.EconomicsConfig = economicsConfig
+				cfg.EpochConfig = epochConfig
+				cfg.GeneralConfig.SovereignConfig = *sovereignExtraConfig
+				cfg.GeneralConfig.VirtualMachine.Execution.WasmVMVersions = []config.WasmVMVersionByEpoch{{StartEpoch: 0, Version: "v1.5"}}
+				cfg.GeneralConfig.VirtualMachine.Querying.WasmVMVersions = []config.WasmVMVersionByEpoch{{StartEpoch: 0, Version: "v1.5"}}
+			},
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, chainSimulator)
+
+	time.Sleep(time.Second)
+
+	chainSimulator.Close()
+}
+
+func TestChainSimulator_GenerateBlocksShouldWork(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	epochConfig, economicsConfig, sovereignExtraConfig, err := LoadSovereignConfigs(sovereignConfigPath)
+	require.Nil(t, err)
+
+	chainSimulator, err := NewSovereignChainSimulator(ArgsSovereignChainSimulator{
+		SovereignExtraConfig: *sovereignExtraConfig,
+		ChainSimulatorArgs: chainSim.ArgsChainSimulator{
+			BypassTxSignatureCheck: false,
+			TempDir:                t.TempDir(),
+			PathToInitialConfig:    defaultPathToInitialConfig,
+			NumOfShards:            1,
+			GenesisTimestamp:       time.Now().Unix(),
+			RoundDurationInMillis:  uint64(6000),
+			RoundsPerEpoch:         core.OptionalUint64{},
+			ApiInterface:           api.NewNoApiInterface(),
+			MinNodesPerShard:       2,
+			ConsensusGroupSize:     2,
+			AlterConfigsFunction: func(cfg *config.Configs) {
+				cfg.EconomicsConfig = economicsConfig
+				cfg.EpochConfig = epochConfig
+				cfg.GeneralConfig.SovereignConfig = *sovereignExtraConfig
+				cfg.GeneralConfig.VirtualMachine.Execution.WasmVMVersions = []config.WasmVMVersionByEpoch{{StartEpoch: 0, Version: "v1.5"}}
+				cfg.GeneralConfig.VirtualMachine.Querying.WasmVMVersions = []config.WasmVMVersionByEpoch{{StartEpoch: 0, Version: "v1.5"}}
```

Comment:
We shouldn't do this for every constructor. This should be done by default on constructor.

Reply:
Also, Why would you need to overwrite the wasm versions, since you use sovereign config anyway, which has those set?

Reply:
Wasm version comes from General Config which is common for both chains and need to be overwritten (same behavior as prefs.toml)